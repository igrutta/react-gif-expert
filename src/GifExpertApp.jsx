import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {
  const [categories, setCategories] = useState([]);

  const onAddCategory = (newCategory) => {
    if (
      categories
        .map((category) => category.toLowerCase())
        .includes(newCategory.toLowerCase())
    ) {
      return;
    }
    setCategories([newCategory, ...categories]);
  };

  return (
    <>
      <h1>GifExpertApp</h1>
      <AddCategory onAddCategory={onAddCategory} />
      {categories.map((category) => (
        <GifGrid key={category} category={category} />
      ))}
    </>
  );
};
