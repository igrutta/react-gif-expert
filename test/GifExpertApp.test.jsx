import { render, screen, fireEvent } from "@testing-library/react";
import { GifExpertApp } from "../src/GifExpertApp";

describe("Pruebas en <GifExpertApp />", () => {
  test("Debe hacer match con el snapshot", () => {
    // run
    const { container } = render(<GifExpertApp />);

    // assert
    expect(container).toMatchSnapshot();
  });

  test("Debe contener el titulo de la aplicación", () => {
    // run
    render(<GifExpertApp />);
    const title = screen.getByRole("heading", { level: 1 });

    // assert
    expect(title.innerHTML).toBe("GifExpertApp");
  });

  test("Debe agregar una nueva categoría", () => {
    // init
    const inputValue = "Homero";

    // run
    render(<GifExpertApp />);

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);

    const category = screen.getByRole("heading", { level: 3 });

    // assert
    expect(category.innerHTML).toBe(inputValue);
  });

  test("No debe agregar una categoría si se repite", () => {
    // init
    const inputValue = "Homero";

    // run
    render(<GifExpertApp />);

    const input = screen.getByRole("textbox");
    const form = screen.getByRole("form");

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);

    fireEvent.input(input, { target: { value: inputValue } });
    fireEvent.submit(form);

    const category = screen.getAllByRole("heading", { level: 3 });

    // assert
    expect(category).toHaveLength(1);
  });
});
