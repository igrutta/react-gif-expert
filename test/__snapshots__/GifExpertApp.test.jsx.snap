// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifExpertApp /> Debe hacer match con el snapshot 1`] = `
<div>
  <h1>
    GifExpertApp
  </h1>
  <form
    aria-label="formulario"
  >
    <input
      placeholder="Buscar gifs"
      type="text"
      value=""
    />
  </form>
</div>
`;
